### ISSUE TITLE
[BUG] no-contradicting-classname rule falsely detects "grid grid-*" classes

### ISSUE THREAD

**Issue by okonomi** [francoismassart/eslint-plugin-tailwindcss#14]
**Describe the bug**

`no-contradicting-classname` rule was detected to no-contradict classes.

**To Reproduce**

1. Write React component:

```
export function App() {
  return <div className="grid grid-cols-3"></div>
}
```

2. run eslint

3. raise detect message

```
error  Classnames grid, grid-cols-3 are conflicting!  tailwindcss/no-contradicting-classname
```

**Expected behavior**

"grid grid-*" classes passed the `no-contradicting-classname` rule.

**Screenshots**

![image](https://user-images.githubusercontent.com/74355/118389015-ae2a7380-b662-11eb-807e-11f91560a644.png)

**Environment (please complete the following information):**

 - OS: macOS Big Sur M1
 - Softwares + version used:
     - VSCode 1.56.1
     - npm v7.11.2, node v16.1.0
     - eslint-plugin-tailwindcss v1.5.4

**Additional context**

**eslint config file or live demo**

sample project is here:

https://github.com/okonomi/eslint-plugin-tailwindcss-sandbox

**Comment by okonomi** [comment]
I tried to fix it, but I'm not sure if it's correct.

https://github.com/okonomi/eslint-plugin-tailwindcss/commit/ac16e2b34be87964887f6bcdb228979f5a34103b

**Comment by francoismassart** [comment]
Hi @okonomi,
thank you for your input, I did try your fork but it break the other tests.

It turns out the regex used for the [`display` properties](https://tailwindcss.com/docs/display) was incorrect.

I did fix it and I'm gonna publish the patch in the `v.1.5.5` release in few minutes.

**Comment by francoismassart** [comment]
@okonomi,
I did a mistake, your proposition was working, it was just the side effects affecting the tests due to the fix.
You can open a PR if you want 👍 
I've been sick these last 2 days and it affects my concentration...

**Comment by okonomi** [comment]
@francoismassart Thank you for the quick fix! It's working well with v1.5.5.

Take care of yourself 🙏
